components/leds.py

```diff
@@ -1,50 +1,67 @@
+import math
 import wpilib
 import random
 import time
-from enum import Enum
-from typing import Optional, Tuple
+from enum import Enum, auto
+from typing import List, Optional, Tuple
+from utilities.scalers import scale_value
 
 
 MAX_BRIGHTNESS = 180  # Between 0-255 of Value on HSV scale
 
 
 class LedColours(Enum):
-    # Use HSV to get nicer fading
+    # Use HSV to get nicer fading, hues are 0-180 so half usual hue
     RED = (0, 255, MAX_BRIGHTNESS)
-    ORANGE = (30, 255, MAX_BRIGHTNESS)
+    ORANGE = (20, 255, MAX_BRIGHTNESS)
+    YELLOW = (30, 255, MAX_BRIGHTNESS)
     PINK = (150, 255, MAX_BRIGHTNESS)
     BLUE = (120, 255, MAX_BRIGHTNESS)
+    CYAN = (90, 255, MAX_BRIGHTNESS)
     GREEN = (60, 255, MAX_BRIGHTNESS)
+    WHITE = (0, 0, MAX_BRIGHTNESS)
     OFF = (0, 0, 0)
 
 
+class DisplayType(Enum):
+    MORSE = auto()
+    PACMAN = auto()
+    RAINBOW = auto()
+    SOLID = auto()
+    PULSE = auto()
+    FLASH = auto()
+
+
+# creates a list of LEDData's from a List of (hsv col, repetitions)
+def make_pattern(
+    data: List[Tuple[LedColours, int]]
+) -> List[wpilib.AddressableLED.LEDData]:
+    pattern_data = []
+    for colour, number in data:
+        for _ in range(number):
+            x = wpilib.AddressableLED.LEDData()
+            x.setHSV(colour.value[0], colour.value[1], colour.value[2])
+            pattern_data.append(x)
```

Comment:
You don't need to mutate these, right?
```suggestion
            x = wpilib.AddressableLED.LEDData()
            x.setHSV(*colour.value)
            pattern_data += [x] * number
```